### ISSUE TITLE
All data gone, without a warning

### ISSUE THREAD

**Issue by vbakke** [devsecopsmaturitymodel/DevSecOps-MaturityModel#334]
I was wondering what the button 'Reset Implemented' was doing.

And whoosh... all my stored data went down the drain. No warning.

**Comment by wurstbrot** [comment]
Hi @vbakke thank you for your feedback.

PRs are welcome.

**Comment by vbakke** [comment]
I can make a PR. I suggest that we move the reset (delete all data) function in under som settings page.

**Comment by vbakke** [comment]
_Ignore this comment that stops botty turning the issue stale_

**Comment by vbakke** [comment]
Just a comment to keep botty happy

**Comment by vbakke** [comment]
Just a comment to keep botty happy

**Comment by vbakke** [comment]
_shush, bot. not stale, just a bit late_

**Comment by PAVAN507-STAR** [comment]
@vbakke Need help as I'm kinda new here ,I have a couple of doubts => 1)what is the use of teams page it kinda looks static & 2) the filters in mapping page when selected filter performed activities it just shows nothing but when both are selected it’s showing all i mean is'nt it like when you select both intersection both filters should come right so when both are selected should'nt it be null set?If these are issues that needs to be resolved I'll raise a PR and work on one of them myself or both if possible and if it's okay.

**Comment by vbakke** [comment]
Hi @PAVAN507-STAR, I'm glad you ask and take an interest. And great new if you are happy providing PRs too. :) 

I've just recently started working with DSOMM so I don't know know the answers, but:
 1) Teams page - Yes, as far as I can see, it is a static page that just displays the teams. _(I'm sure we can display something cool teams related info here later on, though. Do you have any suggestion? : )_
 2) Filters on mapping page - I haven't notice this before. Looks like they are supposed to be filtering on `activity.isImplemented`.  But the the Heatmap page filter on `activity.teamsImplemented` by team. And I don't see any way for a user to set the `activity.isImplemented` flag. Maybe the whole filter should be swapped for the Teams filters, like Circular heatmap (?)

@wurstbrot what do you think?

**Comment by PAVAN507-STAR** [comment]
> Hi [@PAVAN507-STAR](https://github.com/PAVAN507-STAR), I'm glad you ask and take an interest. And great new if you are happy providing PRs too. :)
> 
> I've just recently started working with DSOMM so I don't know know the answers, but:
> 
> 1. Teams page - Yes, as far as I can see, it is a static page that just displays the teams. _(I'm sure we can display something cool teams related info here later on, though. Do you have any suggestion? : )_
> 2. Filters on mapping page - I haven't notice this before. Looks like they are supposed to be filtering on `activity.isImplemented`.  But the the Heatmap page filter on `activity.teamsImplemented` by team. And I don't see any way for a user to set the `activity.isImplemented` flag. Maybe the whole filter should be swapped for the Teams filters, like Circular heatmap (?)
> 
> [@wurstbrot](https://github.com/wurstbrot) what do you think?

1)As per Teams page I think it would be optimal to make the editable as to like to increasing team memebers,teams and stuff like that but the cache comes with we will also have to change the heat map section a bit too.(as per my current knowledge) so i would like someone else more experienced in this repo to guide me a bit on making the pr.
2) I've attached images below as to what the issue is and as you can see it really doesn't make sense atleast in normal way of how filters work.

<img width="1150" alt="Image" src="https://github.com/user-attachments/assets/272f2c26-439e-4028-a40f-aecb7d35a93d" />

<img width="1142" alt="Image" src="https://github.com/user-attachments/assets/168ffc0a-fc71-4982-8af6-5857e68d3a13" />

<img width="1138" alt="Image" src="https://github.com/user-attachments/assets/4151a899-53a3-4f4b-b9e0-a322e1b9e859" />

**Comment by vbakke** [comment]
2) Filters - I'll let @wurstbrot comment on the backstory and direction on further improvement. 


1) Teams - I cannot agree with you more.  It's on my wish list as well.  But the current data model is based on reading the Yaml files.  Any changes are stored in your browser's localStorage. Meaning,  any new teams will only be visible to you.  (And only that same browser.)

So at the moment I've push this change down a bit, on my list.  I think making one common service for reading data (browser local, or server side) is higher priority.  Then it will be easier to change / add features. 

Changing teams is IMHO a very fundamental feature to DSOMM. It is something everyone will need to do. But at the moment,  I think see are better off explaining how to edit the `meta.yaml` (or preferably `team.yaml`) under Usage.  (At least for a little while longer. )

**Comment by wurstbrot** [comment]
Hi,
 
as @vbakke mentioned, teams are fetched from the meta.yaml.
Currently, DSOMM is based on yaml files. A good backend is an option, but requires a lot of effort to be implemented in a good maintainable way.

**Comment by PAVAN507-STAR** [comment]
> Hi,
> 
> as [@vbakke](https://github.com/vbakke) mentioned, teams are fetched from the meta.yaml. Currently, DSOMM is based on yaml files. A good backend is an option, but requires a lot of effort to be implemented in a good maintainable way.
@wurstbrot I would like to collabrate with @vbakke and try to pull of a good maintainable backend with your guidance if it's fine with you sir.

**Comment by vbakke** [comment]
@PAVAN507-STAR , as much as I would love to make a backend to DSOMM, I agree with @wurstbrot . It requires a lot of effort. 

I think, at the moment, there are other improvements that I would put higher on the priority list.

**Comment by PAVAN507-STAR** [comment]
@vbakke Okay ,then we can collaborate on redesigning frontend as you said.

**Comment by vbakke** [comment]
Just created a PR for the original issue in the thread: That locally stored process of teams are deleted without a warning.

@wurstbrot, or others, feel free to suggest better wordings. : )

(@PAVAN507-STAR , regarding the filters on the Matrix I recommend you create a separate issue.)

**Comment by PAVAN507-STAR** [comment]
> Just created a PR for the original issue in the thread: That locally stored process of teams are deleted without a warning.
> 
> [@wurstbrot](https://github.com/wurstbrot), or others, feel free to suggest better wordings. : )
> 
> ([@PAVAN507-STAR](https://github.com/PAVAN507-STAR) , regarding the filters on the Matrix I recommend you create a separate issue.)

Yeah sure will do once I''m done with the other pr i raised

**Comment by vbakke** [comment]
> Hi,
> 
> as [@vbakke](https://github.com/vbakke) mentioned, teams are fetched from the meta.yaml. Currently, DSOMM is based on yaml files. A good backend is an option, but requires a lot of effort to be implemented in a good maintainable way.

We do agree on DSOMM running on yaml files,  @wurstbrot . _But_ DSOMM is also being run from the dsomm.owasp.org site. And it would actually be handy for someone to edit the team names for their localStorage, since they can indeed edit the progress of each of these teams. 

It would make it a lot easier for people to try out DSOMM.  It does make a huge difference to test DSOMM, and show DSOMM to ones managers, if the chart reflected real team names. (I know it is only a psychological difference between seeing `Customer relation team` and team `B`. But from my client meeting experience, it does indeed make much more impact having familiar surroundings. Then one can concentrate of the real issue at hand.)

I will actually suggest that we put _editable teamnames_ on the roeadmap. But only after centralizing the YAML load in a common service. And we should probably make a feature toggle in the meta.yaml, that allows editable team names on dsomm.owasp.org, but maybe not on a local installation.

**Comment by vbakke** [comment]
_This issue is awiting on approval of #355 , stale-bot_
